### PR TITLE
feature: test for serializing and deserializing the events correctly

### PR DESCRIPTION
**Pull request by MikelAlejoBR**
Asserts that the generated JSON by an object mapper can be read back by the parser. The idea is that the contents should not be modified.

### CODE REVIEW

File: src/test/java/com/redhat/cloud/event/parser/ConsoleCloudEventParserTest.java [hunk 1]
```diff
@@ -161,6 +169,57 @@ public void shouldFailOnInvalidJson() throws IOException {
...
+        cloudEvent.setSource(ceSource);
+        cloudEvent.setSpecVersion(ceSpecVersion);
+        cloudEvent.setSubject(ceSubject);
+        cloudEvent.setTime(ceTime);
+        cloudEvent.setType(ceType);
+
+        final String result = objectMapper.writeValueAsString(cloudEvent);
+
+        final ConsoleCloudEventParser consoleCloudEventParser = new ConsoleCloudEventParser();
+        final GenericConsoleCloudEvent<JsonNode> resultCe = consoleCloudEventParser.fromJsonString(result);
```
Comment (MikelAlejoBR) [inline src/test/java/com/redhat/cloud/event/parser/ConsoleCloudEventParserTest.java:210]: This argument is throwing the following exception:

```
java.lang.IllegalArgumentException: argument "in" is null
```

Am I doing something wrong here?

Comment (MikelAlejoBR) [reply src/test/java/com/redhat/cloud/event/parser/ConsoleCloudEventParserTest.java:209]: True... Let me try again. My goal is to check that `LocalDateTime`s get serialized/deserialized as expected, since I'm hitting a weird issue elsewhere, but this is getting in my way...

**Comment by MikelAlejoBR** [comment]
Now, this is the error I was expecting to hit:

```
Caused by: java.time.format.DateTimeParseException: Text '2023-05-09T16:00:02.056626171' could not be parsed at index 29
```

Apparently serializing the Cloud Events using the object mapper makes the `LocalDateTime`s to be serialized with nanoseconds. I guess I am doing something wrong here, @josejulio?

**Comment by josejulio** [comment]
We should update the [DateTimeFormatter](https://github.com/RedHatInsights/event-schemas-java/blob/b99ae08d599ddcfb1d425035b54a22b5f988bd03/src/main/java/com/redhat/cloud/event/parser/modules/Constants.java#L8) reader we have, or even have a fallback - there are some related [tests](https://github.com/RedHatInsights/event-schemas-java/blob/3b8e5ca666fbbdef2a8b93d176745218bd090791/src/test/java/com/redhat/cloud/event/parser/modules/DateTimeModuleTest.java#L16).

I can help updating them, let me know.

**Comment by MikelAlejoBR** [comment]
> We should update the [DateTimeFormatter](https://github.com/RedHatInsights/event-schemas-java/blob/b99ae08d599ddcfb1d425035b54a22b5f988bd03/src/main/java/com/redhat/cloud/event/parser/modules/Constants.java#L8) reader we have, or even have a fallback - there are some related [tests](https://github.com/RedHatInsights/event-schemas-java/blob/3b8e5ca666fbbdef2a8b93d176745218bd090791/src/test/java/com/redhat/cloud/event/parser/modules/DateTimeModuleTest.java#L16).
> 
> I can help updating them, let me know.

I forgot to add the two modules to the object mapper... Everything seems fine for me then, I don't think this test is even necessary, then?

**Comment by josejulio** [comment]
> I forgot to add the two modules to the object mapper... Everything seems fine for me then, I don't think this test is even necessary, then?

I guess you had a valid test before (without the modules)-  By adding the modules you are forcing the json to be without the nanos - and I think it would be nice to accept these values.

But the main issue you had should be fixed by instead of using the ObjectMapper, use the [toJson](https://github.com/RedHatInsights/event-schemas-java/blob/main/src/main/java/com/redhat/cloud/event/parser/ConsoleCloudEventParser.java#L106) method of the parser.

**Comment by MikelAlejoBR** [comment]
> > I forgot to add the two modules to the object mapper... Everything seems fine for me then, I don't think this test is even necessary, then?
> 
> I guess you had a valid test before (without the modules)- By adding the modules you are forcing the json to be without the nanos - and I think it would be nice to accept these values.

I am confused, though... I have added a test to the `DateTimeModuleTest` and it accepts the datetimes with a nanosecond precision —which on the other hand is expected—, yet the parser is complaining about it.

I have also created the following minimal schema:

```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$id": "https://console.redhat.com/api/schemas/events/v1/events.json",
  "title": "console.redhat.com CloudEvents Schema",
  "description": "See https://raw.githubusercontent.com/cloudevents/spec/main/cloudevents/formats/cloudevents.json for base CloudEvents schema. NOTE: this schema inlines and further constrains some CloudEvents properties",
  "type": "object",
  "properties": {
    "time": {
      "type": "string",
      "format": "date-time"
    }
  }
}
```

And used the following payload to validate it:

```json
{
  "time": "2023-05-10T06:07:12.281145701"
}
```

And it validates correctly...

> But the main issue you had should be fixed by instead of using the ObjectMapper, use the [toJson](https://github.com/RedHatInsights/event-schemas-java/blob/main/src/main/java/com/redhat/cloud/event/parser/ConsoleCloudEventParser.java#L106) method of the parser.

Thank you!

File: src/test/java/com/redhat/cloud/event/parser/modules/DateTimeModuleTest.java [hunk 2]
```diff
@@ -97,4 +97,16 @@ public void withNegativeOffsetToReachNextDay() throws JsonProcessingException {
...
+    public void withNanoseconds() throws JsonProcessingException {
+        Helper helper = new Helper();
+        helper.localDateTime = LocalDateTime.parse("2023-05-10T06:02:53.704467873");
+
+        String serialized = objectMapper.writeValueAsString(helper);
+
+        assertEquals("{\"localDateTime\":\"2023-05-10T06:02:53.704467873Z\",\"offsetDateTime\":null}", serialized);
+
+        assertEquals(LocalDateTime.of(2023, 5, 10, 6, 2, 53, 704467873), helper.localDateTime);
+    }
```
Comment (josejulio) [inline src/test/java/com/redhat/cloud/event/parser/modules/DateTimeModuleTest.java:111]: Here you are testing that a LocalDateTime can be serialized - not the other way around which was failing (IIUC)
